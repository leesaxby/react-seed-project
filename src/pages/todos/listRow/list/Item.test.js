import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoItem from './Item';

configure({ adapter: new Adapter() });

const props = {
    item: {
        _id: 1,
        text: 'Item one',
        done: false
    },
    onToggleDone: jest.fn()
};

const wrapper = mount(<TodoItem { ...props }/>);

describe('Todo item component', () => {

    it('Should render todo item with correct text', () => {
        expect( wrapper.find(TodoItem).text() ).toBe('Item one');
    });

    it('Should call onToggleDone when clicked', () => {
        expect(props.onToggleDone.mock.calls.length).toBe(0);

        wrapper.find(TodoItem).simulate('click');

        expect( props.onToggleDone.mock.calls.length ).toBe(1);
    });
});
