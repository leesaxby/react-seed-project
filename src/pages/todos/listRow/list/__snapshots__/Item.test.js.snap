// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An Item  renders component 1`] = `
<li
  aria-labelledby="todoItemButton1Label"
  className="MuiButtonBase-root-12 MuiListItem-root-1 MuiListItem-default-4 MuiListItem-dense-5 MuiListItem-gutters-8 MuiListItem-button-9"
  data-test-id="todo-item"
  id="todoItemButton1Label"
  label="todoItemButton1Label"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  style={
    Object {
      "textDecoration": "",
    }
  }
  tabIndex={0}
>
  <label
    htmlFor="checkboxLabelId"
  >
    <span
      aria-label="todoItemCheckbox1Label"
      className="MuiButtonBase-root-12 MuiIconButton-root-25 MuiPrivateSwitchBase-root-21 MuiCheckbox-root-15 MuiCheckbox-colorSecondary-20"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-30"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-31"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <input
          checked={false}
          className="MuiPrivateSwitchBase-input-24"
          data-indeterminate={false}
          id="todoItemCheckbox1Label"
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
      </span>
    </span>
  </label>
  Item one
  <span
    className="MuiTouchRipple-root-40"
  />
</li>
`;
